google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/gcs/GcsBucketContentEditor.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.intellij.gcs;
+
+import com.intellij.codeHighlighting.BackgroundEditorHighlighter;
+import com.intellij.openapi.fileEditor.FileEditor;
+import com.intellij.openapi.fileEditor.FileEditorLocation;
+import com.intellij.openapi.fileEditor.FileEditorState;
+import com.intellij.openapi.util.Key;
+import java.beans.PropertyChangeListener;
+import javax.swing.JComponent;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Google Cloud Storage {@link FileEditor} implementation. Instantiates the UI panel that allows
+ * browsing of GCS bucket contents.
+ */
+final class GcsBucketContentEditor implements FileEditor {
+
+  private final GcsBucketContentEditorPanel bucketContentPanel;
+  private final GcsBucketVirtualFile bucketVirtualFile;
+
+  GcsBucketContentEditor(@NotNull GcsBucketVirtualFile bucketVirtualFile) {
+    this.bucketVirtualFile = bucketVirtualFile;
+    bucketContentPanel = new GcsBucketContentEditorPanel(bucketVirtualFile);
+  }
+
+  @NotNull
+  @Override
+  public JComponent getComponent() {
+    return bucketContentPanel.getComponent();
+  }
+
+  @Nullable
+  @Override
+  public JComponent getPreferredFocusedComponent() {
+    return bucketContentPanel.getComponent();
+  }
+
+  @NotNull
+  @Override
+  public String getName() {
+    return "gcs-bucket-content";
```

Comment:
What is this name used for? Does it make sense to pull into a constant somewhere else?

Reply:
This is the name that identifies an open editor instance. I'm not 100% sure what its used for internally. Its not used for the name of the editor (the name you see in tab in the gui). its also not used to prevent multiple editors of the same "kind" from being opened (I had to add some logic myself to do this: see GcsBucketPanel#findOpenGcsVirtualFile

I'll just move it to a private constant to make it more visible

Reply:
done